### ISSUE TITLE
Progress reporting in the training job

### ISSUE THREAD

**Issue by leoll2** [open-edge-platform/training_extensions#5199]
We currently do not provide any progress updates from the training loop back to the job, as a result the progress is either 0 or 100.0. There is a special ExecutionContext.report method available within the OTXTrainer, that could be potentially triggered from the callback.

The goal of this ticket is to enable progress reporting in the training job with a reasonably good accuracy, good enough for the user to estimate the remaining job duration.

Since the job has multiple step, we could split the [0-100%] range as follows:
- [0-10%] for dataset preparation
- [10-80%] for the training loop
- [80-95%] for the final evaluation on test set
- [95-100%] for the remaining steps
